### PR TITLE
ios: disable chats in share-sheet based on preferences

### PR DESCRIPTION
**Pull request by levitatingpineapple**

### CODE REVIEW

File: apps/ios/SimpleX SE/ShareView.swift [hunk 1]
```diff
@@ -77,7 +93,7 @@ struct ShareView: View {
             HStack {
                 Group {
                     if #available(iOSApplicationExtension 16.0, *) {
-                        TextField("Comment", text: $model.comment, axis: .vertical)
+                        TextField("Comment", text: $model.comment, axis: .vertical).lineLimit(6)
```
Comment (levitatingpineapple) [inline apps/ios/SimpleX SE/ShareView.swift:96]: Add line limit, so the comment field does not completely obstruct the chat list

File: apps/ios/SimpleXChat/ChatUtils.swift [hunk 2]
```diff
@@ -14,6 +14,53 @@ public protocol ChatLike {
...
+        }
+    }
+
+    public func prohibitedByPref(msgContent: MsgContent?) -> Bool {
+        prohibitedByPref(
+            hasSimplexLink: hasSimplexLink(msgContent?.text),
+            isMediaOrFileAttachment: msgContent?.isMediaOrFileAttachment ?? false,
+            isVoice: msgContent?.isVoice ?? false
+        )
+    }
```
Comment (levitatingpineapple) [inline apps/ios/SimpleXChat/ChatUtils.swift:42]: `func prohibitedByPref(msgContent: MsgContent?)` is potentially redundant.

File: apps/ios/SimpleXChat/ChatUtils.swift [hunk 3]
```diff
@@ -14,6 +14,45 @@ public protocol ChatLike {
     var chatStats: ChatStats { get }
 }
 
+extension ChatLike {
+    public func groupFeatureEnabled(_ feature: GroupFeature) -> Bool {
```
Comment (spaced4ndy) [inline apps/ios/SimpleXChat/ChatUtils.swift:18]: can be reused between this and Chat? it's also ChatLike

Comment (epoberezkin) [reply apps/ios/SimpleXChat/ChatUtils.swift:18]: removed

**Pull request merged**